pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/performance/ConsecutiveLiteralAppendsRule.java

```diff
@@ -74,6 +75,7 @@
         BLOCK_PARENTS.add(ASTIfStatement.class);
         BLOCK_PARENTS.add(ASTSwitchStatement.class);
         BLOCK_PARENTS.add(ASTMethodDeclaration.class);
+        BLOCK_PARENTS.add(ASTTryStatement.class);
```

Comment:
catch and finally should be added too. A simple test-case having 2 catch blocks do different appends / a catch + finally doing appends would trigger these FP too.

Reply:
I'm thinking, we actually may have similar FPs with lambdas, and we should add `ASTLambdaExpression.class` here…

Reply:
I tried to check lambdas. No FPs.